blog/2023-06-16-inline-sdk.mdx

```diff
@@ -0,0 +1,1104 @@
+                videoElement.current.duration) *
+            100;
+        setPlayerState({
+            ...playerState,
+            progress,
+        });
+    };
+
+    const seekTo = (value: number) => {
+        if (value >= 0 && value <= 100) {
+            videoElement.current.currentTime =
+                (videoElement.current.duration / 100) * value;
+            setPlayerState({
+                ...playerState,
+                progress: value,
+            });
+        }
+    };
+
+    const handleVideoSpeed = (event) => {
+        const speed = +event.target.value;
+        videoElement.current.playbackRate = speed;
+        setPlayerState({
+            ...playerState,
+            speed,
+        });
+    };
+
+    const toggleMute = () => {
+        setPlayerState({
+            ...playerState,
+            isMuted: !playerState.isMuted,
+        });
+    };
+
+    useEffect(() => {
+        videoElement.current.muted = playerState.isMuted;
+    }, [playerState.isMuted, videoElement]);
+
+    useEffect(() => {
+        videoElement.current.playbackRate = playerState.speed;
+    }, [playerState.speed, videoElement]);
+
+    useEffect(() => {
+        playerState.isPlaying
+            ? void videoElement.current.play()
+            : void videoElement.current.pause();
+    }, [playerState.isPlaying, videoElement]);
+
+    return {
+        playerState,
+        togglePlay,
+        handleOnTimeUpdate,
+        handleVideoSpeed,
+        toggleMute,
+        seekTo,
+    };
+};
+
+export default useVideoPlayer;
+```
+
+### Creating the component
+
+Now, add the hook to your custom band. Open `src/bands/VideoBand/Band.tsx`. Here you will find a simple react component. This component is the entrypoint for the band. In the `Band` component, add a `player` ref and add your `useVideoPlayer` hook. Add the `props` function parameter to the component.
+
+```tsx title="src/bands/VideoBand/Band.tsx"
+import React, { useRef } from "react";
+import { BandProps } from "inline";
+import "./Band.css";
+import useVideoPlayer from "./useVideoPlayer";
+
+export const Band: React.FC<BandProps> = (props) => {
+    const player = useRef<HTMLVideoElement>();
+    const {
+        playerState,
+        togglePlay,
+        handleOnTimeUpdate,
+        handleVideoSpeed,
+        toggleMute,
+        seekTo,
+    } = useVideoPlayer(player);
+
+    return <div className="video-band">Video Band</div>;
+};
+```
+
+Add a video player and controls in a wrapper in the return statement of the component. The video player will take the url to your video, player ref and the handleOnTimeUpdate function as props to the video player. The url will be accessed using the `band.customBandSettings` prop passed in to the custom band.
+
+:::note
+This custom video band only works with hosted videos, not embedded video players.
+:::
+
+:::note
+Using local video files will require modifications to the SDK. Remember to save your work before making any changes to the SDK.
```

Comment:
I don't know if the audience will understand this note. I think we could just drop it.

Embedding the video in the library is good for a demo, but  it is not a very likely real world use case.

Reply:
Fixed